astropy/io/fits/tests/test_image.py

```diff
@@ -1732,6 +1732,13 @@ def test_compressed_header_missing_znaxis(self):
         with pytest.raises(KeyError):
             comp_hdu.compressed_data
 
+    def test_compressed_header_double_extname(self):
+        """Test that a double EXTNAME with one default value does not
+        mask the non-default value."""
+        with fits.open(self.data('double_ext.fits')) as hdul:
+            assert hdul[1].name == "ccd00"
+            assert "EXTNAME" in hdul[1].header
```

Comment:
Could you also test that if you write the file the COMPRESSED_IMAGE card is removed ?

Reply:
I presumably can't use astropy to read the header though because the code here is going to remove the COMPRESSED_IMAGE card on read even if it was written out.

Reply:
I think the cards would still be in `_header` ?

Reply:
Also, maybe you could add a few more EXTNAME/COMPRESSED_IMAGE to your test file ? (I haven't checked, maybe that's already the case)

Reply:
With the new code I actually strip them from `_header` on read so I can't tell either way. I'm not entirely clear whether `_header` is meant to be treated as read-only since the `CompImageHeader` is clearly designed to write to `_header` for any key that is not one of the special ones.